ui/components/app/assets/token-cell/token-cell.tsx

```diff
@@ -1,133 +1,199 @@
-  amount: number | null,
-  threshold: number,
-  locale: string,
-  options: Intl.NumberFormatOptions,
-): string => {
-  if (amount === null) {
-    return '';
-  }
-  if (amount === 0) {
-    return new Intl.NumberFormat(locale, options).format(0);
-  }
-  return amount < threshold
-    ? `<${new Intl.NumberFormat(locale, options).format(threshold)}`
-    : new Intl.NumberFormat(locale, options).format(amount);
-};
-
 export default function TokenCell({
-  address,
-  image,
-  symbol,
-  chainId,
-  string,
-  tokenFiatAmount,
-  isNative,
+  token,
   privacyMode = false,
   onClick,
 }: TokenCellProps) {
-  const locale = useSelector(getIntlLocale);
-  const currentCurrency = useSelector(getCurrentCurrency);
-  const tokenList = useSelector(getTokenList);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const t = useI18nContext();
   const isEvm = useSelector(getMultichainIsEvm);
-  const erc20TokensByChain = useSelector(selectERC20TokensByChain);
-  const isMainnet = chainId ? isChainIdMainnet(chainId) : false;
-  const tokenData = Object.values(tokenList).find(
-    (token) =>
-      isEqualCaseInsensitive(token.symbol, symbol) &&
-      isEqualCaseInsensitive(token.address, address),
-  );
+  const trackEvent = useContext(MetaMetricsContext);
+  const { safeChains } = useSafeChains();
+  const [showScamWarningModal, setShowScamWarningModal] = useState(false);
 
-  const title =
-    tokenData?.name ||
-    (chainId === '0x1' && symbol === 'ETH'
-      ? 'Ethereum'
-      : chainId &&
-        erc20TokensByChain?.[chainId]?.data?.[address.toLowerCase()]?.name) ||
-    symbol;
+  const decimalChainId = isEvm && parseInt(hexToDecimal(token.chainId), 10);
 
-  const tokenImage =
-    tokenData?.iconUrl ||
-    (chainId &&
-      erc20TokensByChain?.[chainId]?.data?.[address.toLowerCase()]?.iconUrl) ||
-    image;
+  const safeChainDetails: SafeChain | undefined = safeChains?.find((chain) => {
+    if (typeof decimalChainId === 'number') {
+      return chain.chainId === decimalChainId.toString();
+    }
+    return undefined;
+  });
 
-  const secondaryThreshold = 0.01;
-  // Format for fiat balance with currency style
-  const secondary =
-    tokenFiatAmount === null
-      ? undefined
-      : formatWithThreshold(tokenFiatAmount, secondaryThreshold, locale, {
-          style: 'currency',
-          currency: currentCurrency.toUpperCase(),
-        });
+  const tokenDisplayInfo = useTokenDisplayInfo({
+    token,
+  });
 
-  const primary = formatAmount(
-    locale,
-    new BigNumber(Number(string) || '0', 10),
-  );
+  const handleClick = (e?: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    e?.preventDefault();
 
-  const isStakeable = isMainnet && isEvm && isNative;
+    // If the scam warning modal is open, do nothing
+    if (showScamWarningModal) {
+      return;
+    }
 
-  function handleOnClick() {
-    if (!onClick || !chainId) {
+    // Ensure token has a valid chainId before proceeding
```

Comment:
I think that we may want to include a parser in the util file to handle this for multichain. Instead of checking for that the `token.chainId`, we should parse the token identifier, and makes sure that it is a valid namespace before continuing down the component tree.